app/models/shift.rb

```diff
@@ -2,4 +2,26 @@ class Shift < ActiveRecord::Base
   belongs_to :giver, class_name: :User, foreign_key: :giver_id
   belongs_to :taker, class_name: :User, foreign_key: :taker_id
   has_many :offers
+
+  has_one :team, through: :giver, source: :team
+
+  def yes_offers?
+    covered = nil
+    self.offers.each do |offer|
+      covered = true if offer.type == "yes"
+    end
+      covered == true ? self.covered = true : self.covered = false
```

Comment:
You can do this whole method in one line

```ruby
self.covered = offers.any? { |offer| offer.type == 'yes' }
```

You can further improve this by shielding shift from having to know the value of `type` for an accepted offer. If you decide to change the implementation of an accepted offer from a string with value `'yes'` to a boolean with value `true`, it'll be a pain to find all the places you've have `offer.type == 'yes'` to update. 

My suggestion

```ruby
class Offer < ActiveRecord::Base
  ...

  # Feel free to change the name of this method if something else makes more sense
  def accepted?
    type == 'yes'
  end
end

class Shift < AcitveRecord::Base
  ...

  def yes_offers?
    self.covered = offers.any?(&:accepted?)
  end
end